src/hotspot/cpu/aarch64/aarch64_sve.ad

```diff
@@ -1277,40 +1291,36 @@ instruct vstoremaskL(vReg dst, vReg src, vReg tmp, immI_8 size) %{
 
 // load/store mask vector
 
-instruct vloadmask_loadV(vReg dst, vmemA mem) %{
-  predicate(UseSVE > 0 && n->as_Vector()->length_in_bytes() >= 16 &&
-            n->in(1)->bottom_type()->is_vect()->element_basic_type() == T_BOOLEAN);
-  match(Set dst (VectorLoadMask (LoadVector mem)));
-  ins_cost(2 * SVE_COST);
-  format %{ "sve_ld1b $dst, $mem\n\t"
-            "sve_neg $dst, $dst\t # load vector mask (sve)" %}
+instruct loadVMask(vReg dst, vmemA mem) %{
+  predicate(UseSVE > 0);
+  match(Set dst (LoadVectorMask mem));
+  ins_cost(SVE_COST);
+  format %{ "sve_ld1b $dst, $mem\t# load vector mask (sve)" %}
   ins_encode %{
+    // Load mask values which are boolean type, and extend them to the
+    // expected vector element type.
```

Comment:
Isn't this same as existing VectorLoadMaskNode ?

Reply:
I'm not sure whether this can benefit AVX-512 masking load. It loads the values from boolean memory and extends the boolean values to relative other basic types (short,int,long). Different with AVX-512, SVE predicate register controls the vector activity depending on the element type, i.e. it uses 1-bit to control a byte lane, and 2-bits for a short lane, and 4-bits for an int lane, and so on. The type extending/narrowing between real basic type and boolean is always needed. As SVE provides a single instruction to implement the `load+extend` function, we don't need to load values and extend them with external instructions. So that's what the new added node used to do. Maybe a better name for it is `LoadVectorExtendNode`.

Reply:
This new node is performance friendly to SVE. For SVE, we changed the original `VectorLoadMask (LoadVector mem)` to `VectorToMask (LoadVectorMask mem)`. And the same logic to mask store.